### ISSUE TITLE
Hd5 format 

### ISSUE THREAD

**Issue by ikramabdel** [MLearing/Keras-Unet-3d#1]
can we have details on the brats.hd5 to create the same file please ?

**Comment by MLearing** [comment]
Do you want to create your own h5 file to store data? This program is created using the tables library, you can also use h5py to create, you can, you only need to keep the train, label, affine three kinds of data.

**Comment by ikramabdel** [comment]
yeah i wanted to create my own h5 file following the same format so i can test your model ; 
should the order be [Arrays of training data , Arrays of the labels ] ? and another h5 file with the arrays of testing data only ?

**Comment by MLearing** [comment]
There are a total of two h5 files, stored in the same way, their data is stored in the form of a dictionary, such as data["train"], data["label"], data["affine"], which I use h5py library to store Data, you can refer 
with h5py.File(out_file,'w') as f:
        f.create_dataset('train', data=data_storage, compression="gzip")
        f.create_dataset('truth', data=truth_storage, compression="gzip")
        f.create_dataset('affine', data=affine_storage, compression="gzip")
the data_storage is train dataset, You can define the key values of the dictionary data yourself.the other data is the same

**Comment by MLearing** [comment]
You can also store it in npz format as follows:
np.savez_compressed(out_file,train=data_storage, truth=truth_storage, affine=affine_storage)

**Comment by MLearing** [comment]
`def write_data_to_file(training_data_files, out_file, image_shape, truth_dtype=np.uint8, subject_ids=None,normalize=True, crop=True):

    n_samples = len(training_data_files)
    n_channels = len(training_data_files[0]) - 1

    data_storage, truth_storage,affine_storage=[],[],[]
    write_image_data_to_file(training_data_files, data_storage, truth_storage, image_shape,
                             truth_dtype=truth_dtype, n_channels=n_channels,
                             affine_storage=affine_storage, crop=crop)

    data_storage = np.asarray(data_storage,dtype=np.float32)
    truth_storage = np.asarray(truth_storage,dtype=np.float32)
    affine_storage = np.asarray(affine_storage,dtype=np.float32)

    if normalize:
        normalize_data_storage(data_storage)

    print("-" * 90)
    print("start save files!")
    #np.savez_compressed(out_file,train=data_storage, truth=truth_storage, affine=affine_storage)
    with h5py.File(out_file,'w') as f:
        f.create_dataset('train', data=data_storage, compression="gzip")
        f.create_dataset('truth', data=truth_storage, compression="gzip")
        f.create_dataset('affine', data=affine_storage, compression="gzip")


def open_data_file(filename):
    # return np.load(filename)
    return h5py.File(filename,'r')`

**Comment by ikramabdel** [comment]
I'll try that thank you so much !

**Comment by ikramabdel** [comment]
I am still facing some issues , if you can share your h5 data file , i have the brats 2017 dataset please. thank you

**Comment by MLearing** [comment]
you can download dataset in http://learncv.cn/archives/category/projects/3d-u-net
